akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ExternalInteractions.scala

```diff
@@ -157,7 +157,12 @@ private[akka] trait JournalInteractions[C, E, S] {
    * is enabled, old messages are deleted based on `SnapshotCountRetentionCriteria.snapshotEveryNEvents`
    * before old snapshots are deleted.
    */
-  protected def internalDeleteEvents(lastSequenceNr: Long, toSequenceNr: Long): Unit =
+  protected def internalDeleteEvents(lastSequenceNr: Long, toSequenceNr: Long): Unit = {
+    if (setup.isSnapshotOptional) {
+      setup.internalLogger.warn(
+        "Delete events shouldn't be used together with snapshot-is-optional=false. " +
+        "That can result in wrong recovered state if snapshot load fails.")
+    }
```

Comment:
And if we make this method fail if user have enabled the flag? 

A production system configured as such can potentially reach a broken state after event deletion followed by a serialization error.

Reply:
Potentially breaking state, but as far as I can see it, can also not be fail-fast which would feel better if it is to fail at all.

Or maybe it can for the typed API at least?

Reply:
@octonato I have added fail fast for EventSourcedBehavior https://github.com/akka/akka/pull/30238/commits/9898bb65869b2406ca3edd9fcb7d9b145cc3f00d